models.go

```diff
@@ -0,0 +1,145 @@
+package influxdb
+
+import (
+	"sort"
+	"time"
+
+	lp "github.com/influxdata/line-protocol"
+)
+
+// Metric is just a github.com/influxdata/line-protocol.Metric.
+// We alias here to keep abstractions from leaking.
+type Metric = lp.Metric
+
+// RowMetric is a Metric,
+// that has methods to make it easy to add tags and fields
+type RowMetric struct {
+	NameStr string
+	Tags    []*lp.Tag
+	Fields  []*lp.Field
+	TS      time.Time
+}
+
+// TagList returns a slice containing Tags of a Metric.
+func (m *RowMetric) TagList() []*lp.Tag {
+	return m.Tags
+}
+
+// FieldList returns a slice containing the Fields of a Metric.
+func (m *RowMetric) FieldList() []*lp.Field {
+	return m.Fields
+}
+
+// Time is the timestamp of a metric.
+func (m *RowMetric) Time() time.Time {
+	return m.TS
+}
+
+// SortTags orders the tags of a metric alphnumerically by key.
+func (m *RowMetric) SortTags() {
```

Comment:
Tags should probably always be sorted, but I wouldn't add a function.

Reply:
It is a helper so someone making RowMetrics manually can sort their tags easily.  Will add that in the comment.

Reply:
I don't think this should be something the user of the library needs to concern themselves with, we should just do it under the hood.  Is there a reason the client user would want to sort outside of the encoding step?

Reply:
Yes, if they are manually making their own RowMetric.  I can think of lots of reasons to do this, instead of using the Add methods.

Reply:
Why don't we just sort when we encode the data?